src/builder/create_components.rs

```diff
@@ -4,6 +4,15 @@ use serde::Serialize;
 
 use crate::model::prelude::*;
 
+#[derive(Clone, Debug)]
+struct StaticU8<const VAL: u8>;
+
+impl<const VAL: u8> Serialize for StaticU8<VAL> {
+    fn serialize<S: serde::Serializer>(&self, ser: S) -> Result<S::Ok, S::Error> {
+        ser.serialize_u8(VAL)
+    }
+}
+
```

Comment:
Shouldn't uses of this be replaced with `ComponentType`?

Reply:
This avoids having to store the ComponentType in the builder and handle matching on it when serializing. It's a small optimisation.

Reply:
The matching actually isn't a problem because `enum_number!` makes `ComponentType` a newtype around u8, and the variants are actually associated constants. Also, the existing component types already store `ComponentType` inside, so whichever approach we choose we should be consistent.